### DISCUSSION TITLE
MSRV for warnings?

### DISCUSSION THREAD

**Discussion by Kixunil** [rust-bitcoin/rust-bitcoin#1456] [General]
What should be our policy on Rust version (if any) regarding `-D warnings`? IIRC CI currently denies warnings on all versions but sometimes warnings are improved in the compiler and we could e.g. remove `allow` attributes. A recent example: https://github.com/rust-bitcoin/rust-secp256k1/pull/456#discussion_r1045039909

@apoelstra suggested nightly, to me it seems too much of a risk of bringing down our build (nightly does **not** run crater). I suggest stable.

**Comment by tcharding** [comment]
Is there a risk that clippy will warn for something that requires a fix that is not possible with our MSRV? If we assume this is possible using `stable` would reduce the risk as opposed to `nightly`, so unless there is some clear benefit in using `nightly` I'd probably go for `stable`.

We could have a section in the README, or a permanently open good-first-issue, mentioning that we accept nightly clippy fixes even though we only run clippy with X toolchain?

**Reply by Kixunil** [reply]
> Is there a risk

There is always a risk they forget to match up a lint with the MSRV field as it happened historically but I consider the impact minimal.

> `stable` would reduce the risk as opposed to `nightly`

Well, at least it'd be a way to not have whole master broken because the tooling updated.

Note that I *still* want nightly CI but as *informational* - it should not be *required* to pass (we will look at the errors ofc) and if it breaks we report a bug to rust-lang.

>  we accept nightly clippy fixes

It should be "we want nightly clippy fixes for reasonable things".

**Reply by apoelstra** [reply]
>There is always a risk they forget to match up a lint with the MSRV field as it happened historically but I consider the impact minimal.

To be clear, @tcharding if you weren't aware, we can tell clippy about our MSRV and it'll do the right thing.

**Reply by tcharding** [reply]
> we can tell clippy about our MSRV and it'll do the right thing.

Of course, I forgot that, we do that already.
